.github/workflows/publish.yml

```diff
@@ -0,0 +1,70 @@
+name: Publish packages to npm
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: "publish"
+  cancel-in-progress: false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    strategy:
+      max-parallel: 1 # has to be 1 (one) as there are dependencies between the published packages
+      matrix:
+        package: [ base, server, client, client-react ]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '22.x'
+          registry-url: 'https://registry.npmjs.org'
+          scope: '@krmx'
+      - name: Publish Packages
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          cd ${{ matrix.package }}
+          echo "-- @krmx/${{ matrix.package }} --"
+
+          LOCAL_VERSION=$(node -e "console.log(require('./package.json').version);")
+          echo "Local version: $LOCAL_VERSION"
+
+          REMOTE_VERSION=$(npm show "@krmx/${{ matrix.package }}" version)
+          echo "Remote version: $REMOTE_VERSION"
+
+          if [ "$LOCAL_VERSION" = "$REMOTE_VERSION" ]; then
```

Comment:
Instead of matching, it should be a greater than comparison in semver. Otherwise older version will try to publish